tallycash/extension
# 🕵️‍♂️ Show me the money: Query all tokens on `tokenRefreshes` alarm

The asset discovery should happen on new account addition
through the `IndexingService.retrieveTokenBalances()` or
the `ChainService.loadAssetTransfers()`.

At the moment the `ChainService.loadAssetTransfers()` is broken
for an unknown reason and this result in us not picking up any
new tokens, that were not present at the time of adding the new
address.

Querying for all the tokens should not be a costly addition to
out api usage — haven't checked though — but will provide
us a solution for now and a safety net for the future.

## Next steps
- fix the `loadAssetTransfers` flow
- implement [event listener on provider](https://docs.ethers.io/v5/api/providers/provider/#Provider--events) (and move away from polling)


## Test
- import a wallet and send an untracked asset to it from another wallet/address
- wait max 1 minute
- see the new asset appear

Closes #1158 #1082
